### DISCUSSION TITLE
Signing solana messages

### DISCUSSION THREAD

**Discussion by Rimantovas** [ethereum/web3.js#7167] [Q&A]
How can you sign messages with solana?

**Comment by TatyOko28** [comment]
To sign messages with Solana, you use a Solana Keypair or a Wallet (like Phantom) and sign a message using the signMessage function from the Solana Web3.js SDK.

 If using a Keypair (Backend Signing):
Use signMessage() with a Solana Keypair.
 If using Phantom Wallet (Frontend Signing):
Use signMessage() from window.solana for user-approved signing.
